### ISSUE TITLE
custom validation rule

### ISSUE THREAD

**Issue by mhndev** [Respect/Validation#653]
I want to write new custom rule but actually it's not possible.
```
!$reflection->isSubclassOf('Respect\\Validation\\Validatable')
```
why should you write such code ??
is it really important that all rules should have your namespace ??

I need to add a custom rule called 'unique'.
what should I do ?

and also in your Validator class you've instantiated Factory class .
```
    protected static function getFactory()
    {
        if (!static::$factory instanceof Factory) {
            static::$factory = new Factory();
        }

        return static::$factory;
    }

```

this is called hard code.
at least you could get Factory object from constructor.

**Comment by batusa** [comment]
Hi @mhndev ,

Your code should not be part of the namespace's library. It should just implements `Validatable` interface. If you have your stand-alone code and don't wanna touch it, you can create an Adapter. This Adapter would use your code and implements `Validatable` interface in order to be used as a rule by Validation.

Regarding the Factory object, there's a setter method to inject it.

**Comment by batusa** [comment]
I thought I had put a link to the docs, here it is: http://respect.github.io/Validation/docs/

In the section **`Custom rules`** it shows you how to create a custom rule. if it's not clear enough, let me know.

**Comment by alganet** [comment]
There is also the quick and dirty `v::callback` that you can use for very special cases:

```
v::callback(function ($input) { return true; });
v::callback([$this, 'methodName']);
```

**Comment by mhndev** [comment]
thank you I already wrote my custom event using:
`
v::with()
`
but I have another problem which I have solved in a tricky and hacky way and I want to know the true solution.
and my problem is consider I have a register form which has multiple inputs such as mobile and email and ... which none of them are required (only username and password are).
but I want when user fill in email input the value should be validated against invalid email address.
